src/main/java/madstax/model/util/ResourceIO.java

```diff
@@ -38,9 +39,9 @@ public static ImageIcon getImageIcon(String imageName) {
 
     public static <T> List<T> parseDataFromCSV(String fileName, Class<? extends T> type) {
         String csvFileRelativePath = String.format("csv/%s.csv", fileName);
-        InputStream csvStream = getResourceStream(csvFileRelativePath);
 
         try {
+            InputStream csvStream = (type == Teacher.class) ? getResourceStream(csvFileRelativePath) : new FileInputStream(csvFileRelativePath);
```

Comment:
Don't suppose there is a more readable way of writing this?  because I have literally no idea what is going on here hahaha but that must just be me ;)

Reply:
http://tutorials.jenkov.com/java/ternary-operator.html

Reply:
thanks 👍